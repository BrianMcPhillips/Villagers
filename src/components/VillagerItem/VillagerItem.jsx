import React from 'react';
import PropTypes from 'prop-types';
import styles from './VillagerItem.module.css';

const VillagerItem = ({ name, imageUrl }) => (
  <div className={styles.Villager}>
    <figure>
      <img src={imageUrl} alt={name} />
      <figcaption>{name}</figcaption>
    </figure>
  </div>
);

VillagerItem.propTypes = {
  name: PropTypes.string.isRequired,
  imageUrl: PropTypes.string.isRequired
};

export default VillagerItem;
