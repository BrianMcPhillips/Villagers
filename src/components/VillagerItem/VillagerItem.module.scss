.Villager {
  width: 5rem;
  img {
    padding: 0;
    max-width: 5rem;
  };
  figcaption {
    text-align: center;
  };
}
