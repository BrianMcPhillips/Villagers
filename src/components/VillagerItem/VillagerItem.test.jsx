import React from 'react';
import { render, cleanup } from '@testing-library/react';
import VillagerItem from './VillagerItem';

describe('VillagerItem component', () => {
  afterEach(() => cleanup());
  it('renders VillagerItem', () => {
    const { asFragment } = render(<VillagerItem 
      imageUrl="https://placeCage.com/200/200"
      name="Sir Nick Cage"
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
