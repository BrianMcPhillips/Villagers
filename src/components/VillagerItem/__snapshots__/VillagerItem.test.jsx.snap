// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VillagerItem component renders VillagerItem 1`] = `
<DocumentFragment>
  <div
    class="Villager"
  >
    <figure>
      <img
        alt="Sir Nick Cage"
        src="https://placeCage.com/200/200"
      />
      <figcaption>
        Sir Nick Cage
      </figcaption>
    </figure>
  </div>
</DocumentFragment>
`;
